### ISSUE TITLE
bug: waybar is missing gtk wrapping with systemd integration

### ISSUE THREAD

**Issue by colemickens** [nix-community/home-manager#2793]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

I'm doing some fixing up of some sway-related stuff in HM and noticed that waybar's gtk seems to ignore my settings when run under systemd.

### System information

```markdown
I'm on nixos-unstable, that command doesn't print anything for me.
```

**Comment by colemickens** [comment]
actually, it seems like everything is suffering from that, including portal-launched stuffs.

**Comment by colemickens** [comment]
The GTK theme, for example. It's a bit unreliable though, yesterday I had it working a few times as well. I need to look into this more and add some more details.

**Comment by berbiche** [comment]
You could try changing the GTK theme with `dconf write /org/gnome/desktop/interface/gtk-theme "'Adwaita'"` or `gsettings set org.gnome.desktop.interface gtk-theme Adwaita`.
Waybar's interface should automatically update.

**Comment by colemickens** [comment]
Yes, I have the dconf in there, and I use the gtk.nix module as well. It seems like its working fine, I might've had some local state gtk.css or something interfering. I'll close for now.

(or it may've been one of the other little things I've fixed up elsewhere) closing for now.

**Comment by colemickens** [comment]
okay, yeah, my gsettings script is failing, it can't find the desktop schemas, so the gsettings commands aren't doing anything. I feel like I had fixed this somehow ages ago...

**Comment by colemickens** [comment]
yeah, I had disabled the gtk wrapper somehow. I think I got confused and thought the base wrapper was disabled (since hm/nixos has user dbus socket activation now), but in fact it's enabled by default and I still needed to opt into the gtk wrapper.

**Comment by berbiche** [comment]
Is there anything actionable on Home Manager's side?

**Comment by uningan** [comment]
I have this same problem with every GTK3 app, GTK4 working well, when trying to change theme on the fly.

Im running user session with `wayland.windowManager.sway.wrapperFeatures.gtk = true;`

And using portals in configuration.nix:
```
  xdg.portal = {
    enable = true;
    gtkUsePortal = true;
    extraPortals = [ pkgs.xdg-desktop-portal-gtk ];
    wlr.enable = true;
  };
```
I then run  `gsettings set org.gnome.desktop.interface gtk-theme Adwaita-dark`. But gtk3 apps ignore new theme.

I open an issue in nixpkgs https://github.com/NixOS/nixpkgs/issues/168485 but maybe is preferred to follow here.

**Comment by JavaCafe01** [comment]
I don't know if this is part of the issue, but icons in the taskbar are not found for apps when systemd integration is enabled.

For example, here:

![image](https://user-images.githubusercontent.com/33443763/201835453-3beb68b5-b211-4df8-a307-ab165b79c717.png)

The three apps are firefox, wezterm, and wezterm. When running waybar normally without systemd integration I get the app icons. With the integration, I get the picture above.

**Comment by Cu3PO42** [comment]
The above issue experienced by @javacafe01 can be worked around by importing `PATH` into the systemd user environment, either via `systemctl --user import-environment PATH` or `dbus-update-activation-environment --systemd PATH` or preferably the systemd integration of your favorite DE in HM.
